be/src/Project/Web/Buyer/Buyer.Web/Areas/Products/Services/Products/ProductsService.cs

```diff
@@ -15,31 +15,53 @@
 using Foundation.PageContent.Definitions;
 using Foundation.Extensions.ExtensionMethods;
 using Foundation.Media.Services.MediaServices;
+using Buyer.Web.Areas.Products.Services.CompletionDates;
+using IdentityServer4.Extensions;
 
 namespace Buyer.Web.Areas.Products.Services.Products
 {
     public class ProductsService : IProductsService
     {
-        private readonly IProductsRepository productsRepository;
-        private readonly IMediaService mediaService;
-        private readonly IOptions<AppSettings> options;
-        private readonly LinkGenerator linkGenerator;
+        private readonly IProductsRepository _productsRepository;
+        private readonly IMediaService _mediaService;
+        private readonly IOptions<AppSettings> _options;
+        private readonly LinkGenerator _linkGenerator;
+        private readonly ICompletionDatesService _completionDatesService;
 
         public ProductsService(
             IProductsRepository productsRepository,
             IMediaService mediaService,
             IOptions<AppSettings> options,
-            LinkGenerator linkGenerator)
+            LinkGenerator linkGenerator,
+            ICompletionDatesService completionDatesService)
         {
-            this.productsRepository = productsRepository;
-            this.mediaService = mediaService;
-            this.options = options;
-            this.linkGenerator = linkGenerator;
+            _productsRepository = productsRepository;
+            _mediaService = mediaService;
+            _options = options;
+            _linkGenerator = linkGenerator;
+            _completionDatesService = completionDatesService;
+        }
+
+        public async Task<Product> GetProductAsync(string token, string language, Guid? productId, Guid? sellerId)
+        {
+            var product = await _productsRepository.GetProductAsync(productId, language, token);
+
+            if (product is not null)
+            {
+                if (_options.Value.CompletionDatesUrl.IsNullOrEmpty() is false)
+                {
+                    product = (await _completionDatesService.GetCompletionDatesAsync(token, language, sellerId, new List<Product> { product })).FirstOrDefault();
```

Comment:
suggestion: you should only assign product.CompletionDate here. It makes also no sense to wrap one product into the list, please either simplify or make another version of GetCompletionDatesAsync function